src/optimagic/logging/types.py

```diff
@@ -0,0 +1,189 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Literal
+
+from optimagic.typing import (
+    DictLikeAccess,
+    OptimizationType,
+    OptimizationTypeLiteral,
+    PyTree,
+)
+
+
+class StepStatus(str, Enum):
+    """Status of a step in a process.
+
+    Attributes:
+        SCHEDULED: Indicates that the step is scheduled but not yet started.
+        RUNNING: Indicates that the step is currently in progress.
+        COMPLETE: Indicates that the step has completed successfully.
+        SKIPPED: Indicates that the step was skipped.
+
+    """
+
+    SCHEDULED = "scheduled"
+    RUNNING = "running"
+    COMPLETE = "complete"
+    SKIPPED = "skipped"
+
+
+StepStatusLiteral = Literal["scheduled", "running", "complete", "skipped"]
+
+
+class StepType(str, Enum):
+    """Type of step in a process.
+
+    Attributes:
+        OPTIMIZATION: Represents an optimization step.
+        EXPLORATION: Represents an exploration step.
+
+    """
+
+    OPTIMIZATION = "optimization"
+    EXPLORATION = "exploration"
+
+
+StepTypeLiteral = Literal["optimization", "exploration"]
+
+
+class ExistenceStrategy(str, Enum):
+    """Strategies to handle the existence of a database or table.
+
+    Attributes:
+        RAISE: Raise an error if the database resp. table exists.
+        EXTEND: Extend the existing database or table.
+        REPLACE: Replace the existing database or table.
+
+    """
+
+    RAISE = "raise"
+    EXTEND = "extend"
+    REPLACE = "replace"
+
+
+ExistenceStrategyLiteral = Literal["raise", "extend", "replace"]
+
+
+@dataclass(frozen=True)
+class CriterionEvaluationResult(DictLikeAccess):
```

Comment:
Maybe rename to `OptimizerState`, `IterationState` or `OptimizerIterationState`; Actually I think you suggested that at some point and I like it. `CriterionEvaluationResult` is too narrow. Then there might be follow-up renamings (e.g. `iteration_store` to `state_store`

Reply:
Agree, will do